src/foam/dao/ArrayDAO.js

```diff
@@ -29,6 +29,14 @@ foam.CLASS({
 
   properties: [
     {
+      class: 'Class',
+      name: 'of',
+      factory: function() {
+        if (this.array.length === 0) return foam.lookup('foam.core.FObject');
```

Comment:
or not: https://travis-ci.org/foam-framework/foam2/jobs/228443299. What's the right way to pull in the `lookup()` function from the context?